src/auth/crypto.cpp

```diff
@@ -1,18 +1,31 @@
-// Copyright 2022 Memgraph Ltd.
+// Copyright 2023 Memgraph Ltd.
 //
 // Licensed as a Memgraph Enterprise file under the Memgraph Enterprise
 // License (the "License"); by using this file, you agree to be bound by the terms of the License, and you may not use
 // this file except in compliance with the License. You may obtain a copy of the License at https://memgraph.com/legal.
 //
 //
-
 #include "auth/crypto.hpp"
 
+#include <gflags/gflags.h>
 #include <libbcrypt/bcrypt.h>
+#include <openssl/sha.h>
 
 #include "auth/exceptions.hpp"
 
+inline constexpr std::string_view default_password_encryption = "bcrypt";
+inline constexpr std::string_view sha256_password_encryption = "sha256";
+inline constexpr std::string_view sha256_1024_iterations_password_encryption = "sha256-1024";
+
+inline constexpr uint64_t ONE_SHA_ITERATION = 1;
+inline constexpr uint64_t MULTIPLE_SHA_ITERATIONS = 1;
+
+// NOLINTNEXTLINE(cppcoreguidelines-avoid-non-const-global-variables)
+DEFINE_string(password_encryption_algorithm, default_password_encryption.data(),
+              "The password encryption algorithm used for authentication");
```

Comment:
We should put the options into the flag description, does GFlags has some way of specifying an enum? So, it's less error prone if we add more options?

Reply:
Maybe somehow with the flag validation logic -> https://github.com/memgraph/memgraph/blob/master/src/utils/flag_validation.hpp

Reply:
💪